### ISSUE TITLE
Reduce deployment fees for new contracts to 1 GAS on Testnet

### ISSUE THREAD

**Issue by lock9** [neo-project/neo#2965]
**Summary or problem description**
We need to facilitate testnet deployments. This will facilitate integration and testing with other Apps that support testnet (neofs, wallets etc). The existing faucet offers users 50 GAS. This amount is enough to deploy the contract only 4 times (not to mention the Faucet, but this is another problem)

**Do you have any solution you want to propose?**
Reduce the GAS required to deploy smart contracts on Testnet to 1 GAS. 

**Neo Version**
- Neo 3

**Where in the software does this update apply to?**
- Network Policy

**Comment by cschuchardt88** [comment]
I think it would be easier to just increase the rewards to 1000 GAS and keep it at 50 NEO

**Comment by Liaojinghui** [comment]
@steven1227

**Comment by lock9** [comment]
> I think it would be easier to just increase the reward to 1000 GAS and keep it at 50 NEO

The problem is that we can't generate GAS if we have to. Also, the deployment fee can be easily changed using the existing governance system.

**Comment by roman-khimov** [comment]
Against. Faucet can give you GAS every day (maybe can be tuned to 100 GAS, but it's a separate matter) and contract updates (new deployments are not needed in many cases) do not require this much GAS. Symmetric mainnet/testnet configuration is a bit easier for people to handle.

**Comment by cschuchardt88** [comment]
@lock9 I mean for the faucet, like @roman-khimov said. Big projects may require more than 10 deploys of different contracts. That's why I said 1000 GAS, So that way they can also send gas to various multi-sig address or contracts for testing.

**Comment by Liaojinghui** [comment]
> @lock9 I mean for the faucet, like @roman-khimov said. Big projects may require more than 10 deploys of different contracts. That's why I said 1000 GAS, So that way they can also send gas to various multi-sig address or contracts for testing.

For projects that require more GAS, i think they can send email to ngd asking for that.

**Comment by lock9** [comment]
> Symmetric mainnet/testnet configuration is a bit easier for people to handle

I agree. However, neo-express default setup is already different from the mainnet and testnet. Developers need to deal with that all the time. Maybe we can make neo-express use the same fees instead.

> contract updates (new deployments are not needed in many cases) do not require this much GAS.

The problem here is that developers usually don't use the update feature:
<img width="1321" alt="Screenshot 2023-11-13 at 12 35 52" src="https://github.com/neo-project/neo/assets/38396062/cbfc656f-bb0a-4f85-bfd7-cbbbd49e0b29">

All of these are the same contract/developer:
<img width="1315" alt="Screenshot 2023-11-13 at 12 36 42" src="https://github.com/neo-project/neo/assets/38396062/f2166231-bb42-4d90-b347-03a5e7d93037">

What else can we do to allow them to deploy these contracts? One option is to try to educate them to use the update feature. We are doing this on the extension but it will likely be better if we had a more holistic approach.

**Comment by lock9** [comment]
>Against. Faucet can give you GAS every day (maybe can be tuned to 100 GAS, but it's a separate matter) and contract updates (new deployments are not needed in many cases) do not require this much GAS. Symmetric mainnet/testnet configuration is a bit easier for people to handle.

You are right. I've updated the proposed solution.

**Comment by shargon** [comment]
I prefer to send 100 NEO and 100 GAS, if they vote, they can get more gas, and they will start voting.

**Comment by lock9** [comment]
I just saw this on Discord:

<img width="817" alt="Screenshot 2023-11-14 at 11 37 10" src="https://github.com/neo-project/neo/assets/38396062/21ae6506-4245-4155-b46e-6fe3b7b6a685">

We need to facilitate Testnet GAS access ASAP. It's a major blocker for many users.

**Comment by cschuchardt88** [comment]
>@Liaojinghui For projects that require more GAS, i think they can send email to ngd asking (apply) for that.

https://neo.org/testcoin/apply

**Comment by Liaojinghui** [comment]
> > @Liaojinghui For projects that require more GAS, i think they can send email to ngd asking (apply) for that.
> 
> https://neo.org/testcoin/apply

Well, I am not suprised neo have this webpage, i am suprised that you can find this webpage. I did not find the entry point in the neo website.

**Comment by steven1227** [comment]
> I think faucet can be changed to 1000 right? Won't hurt since it's testnet. @steven1227

I can adjust the faucet with no problem. But for development, I use local env more than testnet. I do not know what scenario that user has to deploy contracts 10 times a day.

**Comment by roman-khimov** [comment]
Keep in mind that testnet is a public network as well. GAS controls resources of this network. Giving out too much GAS can increase risks of network abuse. Yeah, it's a testnet, but we're usually not very excited when it breaks.

**Comment by Liaojinghui** [comment]
> Keep in mind that testnet is a public network as well. GAS controls resources of this network. Giving out too much GAS can increase risks of network abuse. Yeah, it's a testnet, but we're usually not very excited when it breaks.

For this,  I can not agree, blockchain is supposed to be used by people, if we have to worry about network abuse, how could we find issues in the network.I would say, Ok, go ahead and use the testnet however you want, if you break the testnet, very good, we find a problem.

**Comment by lock9** [comment]
>I can adjust the faucet with no problem. But for development, I use local env more than testnet. I do not know what scenario that user has to deploy contracts 10 times a day.

Sometimes, you need to deploy multiple contracts. The Flamingo team seems to deploy their contract multiple times a day. Also, 50 GAS allows you to deploy 4 contracts, not 5.
